asyncpg/connection.py

```diff
@@ -314,7 +339,11 @@ async def execute(self, query: str, *args, timeout: float=None) -> str:
         self._check_open()
 
         if not args:
-            return await self._protocol.query(query, timeout)
+            start = time.monotonic()
```

Comment:
I think a context manager would probably be nicer/reduce duplication.

Reply:
Agreed, change incoming.